libs/hdf-converters/src/conveyor-mapper.ts

```diff
@@ -0,0 +1,244 @@
+  const newout: Record<string, unknown>[] = [];
+  _.forEach(res, (value) => {
+    const temp = value as Record<string, unknown>;
+    const temp2: string = _.get(value, 'sha256') || '';
+    _.set(temp, 'filename', _.get(mapped, temp2));
+    const descriptions = _.map(
+      _.get(temp, 'result.sections') as Record<string, unknown>[],
+      (val) =>
+        createDescription(
+          val as Record<string, unknown>,
+          _.get(temp, 'result.score') as number,
+          _.get(temp, 'response.milestones.service_started') as string,
+          _.get(temp, 'response.service_name') as string
+        )
+    );
+    if (descriptions.length == 0) {
+      descriptions.push({
+        status: ExecJSON.ControlResultStatus.Passed,
+        code_desc: 'NA',
+        start_time: _.get(temp, 'response.milestones.service_started') as string
+      });
+    }
+    _.set(temp, 'result.sections', descriptions);
+    newout.push(temp);
+  });
+  const groups = _.groupBy(newout, (value) => {
+    return _.get(value, 'response.service_name');
+  });
+  return groups;
+}
+
+function controlMappingConveyor(): MappedTransform<
+  ExecJSON.Control & ILookupPath,
+  ILookupPath
+> {
+  return {
+    id: {path: 'sha256'},
+    title: {path: 'filename'},
+    desc: {path: 'cve_summary'},
+    impact: {
+      path: 'result.score',
+      transformer: (value) => {
+        return value / 1000;
+      }
+    },
+    refs: [],
+    tags: {
+      nist: DEFAULT_NIST_TAG
+    },
+    source_location: {
+      ref: ''
+    },
+    results: [
+      {
+        path: 'result.sections',
+        status: {path: 'status'},
+        code_desc: {path: 'code_desc'},
+        start_time: {path: 'start_time'}
+      }
+    ]
+  };
+}
+
+export class ConveyorMapper extends BaseConverter {
+  data: Record<string, unknown>;
+  type: string;
+  defaultMapping(): MappedTransform<ExecJSON.Execution, ILookupPath> {
+    return {
+      platform: {
+        name: 'Heimdall Tools',
+        release: HeimdallToolsVersion
+      },
+      version: HeimdallToolsVersion,
+      statistics: {},
+      profiles: [
+        {
+          name: this.type,
+          version: {path: 'api_server_version'},
+          title: {path: 'api_response.params.description'},
+          supports: [],
+          attributes: [],
+          groups: [],
+          status: 'loaded',
+          controls: [
+            {
+              path: 'api_response.results',
+              ...controlMappingConveyor()
+            }
+          ],
+          sha256: ''
+        }
+      ]
+    };
+  }
+  constructor(
+    conveyor: Record<string, unknown>,
+    meta: Record<string, unknown>,
+    type: string
+  ) {
+    const data = meta;
+    _.set(data, 'api_response.results', conveyor);
+    super(data);
+    this.type = type;
+    this.data = data;
+    this.setMappings(this.defaultMapping());
+  }
+}
+
+export class ConveyorResults {
+  data: unknown;
+  meta: Record<string, unknown>;
```

Comment:
since you're basing this off of the asff mapper structure, then i would combine meta into the data field unless you're having additional information come from outside the results file.  for the asff mapper, it was possible to pass in additional context that could better tune the results for some of the specialcasings, but i don't think that's what's happening here.  based off of talking with you, it's mostly metadata information from within the results file itself which we could just pass into the data field and use where necessary with the typical path/recursive structure instead of grabbing it from some arbitrary variable.

Reply:
removed